sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -433,7 +465,7 @@ def _collect_plans(self, rows, cursor, deadline):
                             "collection_errors": collection_errors,
                         },
                         "query_signature": row['query_signature'],
-                        "statement": row['text'],
+                        "statement": row['statement_text'],
```

Comment:
If it's a "procedure sample", where the plan represents an entire procedure with multiple statements in it, then should the statement text also represent the entire procedure definition? It looks like in this case it'll just be one statement from within that procedure which happened to reach this point in the code first? 

Also which `query_signature` are we tagging this procedure plan with? We need to make sure we're still able to look up the procedure plan for all statements contained within that procedure.

Reply:
> If it's a "procedure sample", where the plan represents an entire procedure with multiple statements in it, then should the statement text also represent the entire procedure definition? It looks like in this case it'll just be one statement from within that procedure which happened to reach this point in the code first?

yeah I wasn't sure about this

> Also which query_signature are we tagging this procedure plan with? We need to make sure we're still able to look up the procedure plan for all statements contained within that procedure.

it would use the query_signature we calc from the `statement_text`... we could also calc the signature for the overall procedure `text` and add that as a new field in the event. As I mentioned above, the `query_plan_hash` is not always the same for queries which are contained within the same procedure

Reply:
OK going to send the full plan text as the `statement` field, which is indexed and will allow customers to search for it in query samples

also calc the query signature for the full plan text and add a new field on the event called `procedure_signature` && tag all metrics with that, which can be used to relate query metrics to their full plans in the query panel 

TODO: test a plan that takes parameters with the same internal queries, and see if that produces the same plan handle. This should be helpful in determining if our new sampling strategy makes sense.

Reply:
06c35d82aa48a56a1fe4f742053230f087f0293c contains changes we talked about && a test for procedures that take parameters. Looks like plan handle does not change so we should be good with this sampling strategy